### PR TITLE
Added authentication module with login route

### PR DESCRIPTION
**Pull request by yashodgayashan**

### CODE REVIEW

File: BackendNew/controllers/authentication.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+              }
+            }
+          );
+        } else {
+          res.status(200).send(generateJwtToken(result.data[0]));
+        }
+      }
+    });
+  }
+};
```
Comment (hasinichathu) [inline BackendNew/controllers/authentication.js:73]: What's the reason to check student table first? Is there any specific reason? If Search is other way round first admin then company lastly student then can't we reduce processing cost. coz admin has less entries whereas student table has more entries.

File: BackendNew/models/company.js [hunk 2]
```diff
@@ -15,3 +15,15 @@ exports.hasCompany = (companyName, callback) => {
...
 };
+
+exports.getCompanyByUserNamePassword = (companyId, password, callback) => {
+  var sqlString =
+    "SELECT companyId as Id, RoleName as Role, Name FROM Company WHERE companyId = ? AND password = ?";
+  sql.query(sqlString, [companyId, password], (err, result) => {
+    if (err) {
+      callback(err, null);
+    } else {
+      callback(null, { data: result });
```
Comment (hasinichathu) [inline BackendNew/models/company.js:26]: I have doubt whether it is ok to send password between functions

File: BackendNew/controllers/authentication.js (hunk 1 above)
Comment (yashodgayashan) [reply BackendNew/controllers/authentication.js:73]: Yeah I though it. I will change it.

File: BackendNew/controllers/authentication.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+generateJwtToken = user => {
+  const payload = { Name: user.Name, Id: user.Id, Role: user.Role };
```
Comment (hasinichathu) [inline BackendNew/controllers/authentication.js:76]: For frontend processing I need to have whether user is admin, student or company. But from this "Role: user.Role" several user roles can be obtained. eg:- super-admin, admin etc

Comment (yashodgayashan) [reply BackendNew/controllers/authentication.js:76]: yes. What are the roles you need?

Comment (hasinichathu) [reply BackendNew/controllers/authentication.js:76]: If you can just return table name, that's enough.

**Comment by yashodgayashan** [comment]
@hasinichathu Resolved requested changes.

**Pull request merged**